pkg/accounting/accounting.go

```diff
@@ -299,6 +302,50 @@ func (a *Accounting) Debit(peer swarm.Address, price uint64) error {
 	accountingPeer.lock.Lock()
 	defer accountingPeer.lock.Unlock()
 
+	cost := price
+	// see if peer has surplus balance to deduct this transaction of
+	surplusBalance, err := a.SurplusBalance(peer)
+	if err != nil {
+		if !errors.Is(err, ErrPeerNoBalance) {
+			return fmt.Errorf("failed to get surplus balance: %w", err)
+		}
+	}
+	if surplusBalance > 0 {
+
+		// get new surplus balance after deduct
+		newSurplusBalance, err := subtractI64mU64(surplusBalance, price)
+		if err != nil {
+			return err
+		}
+
+		// if nothing left for debiting, store new surplus balance and return from debit
+		if newSurplusBalance >= 0 {
+			err = a.store.Put(peerSurplusBalanceKey(peer), newSurplusBalance)
+			if err != nil {
+				return fmt.Errorf("failed to persist surplus balance: %w", err)
+			}
+			return nil
```

Comment:
what about metrics in that case? `TotalDebitedAmount` and `DebitEventsCount`. Technically this is still a debit operation.